homeassistant/components/shelly/light.py

```diff
@@ -120,30 +119,25 @@ def async_setup_rpc_entry(
         return
 
     if light_key_ids := get_rpc_key_ids(coordinator.device.status, "light"):
-        # Light mode remove RGB & RGBW entities, add light entities
-        async_remove_shelly_rpc_entities(
-            hass, LIGHT_DOMAIN, coordinator.mac, ["rgb:0", "rgbw:0"]
-        )
         async_add_entities(RpcShellyLight(coordinator, id_) for id_ in light_key_ids)
-        return
 
-    light_keys = [f"light:{i}" for i in range(SHELLY_PLUS_RGBW_CHANNELS)]
+    if cct_key_ids := get_rpc_key_ids(coordinator.device.status, "cct"):
+        async_add_entities(RpcShellyCctLight(coordinator, id_) for id_ in cct_key_ids)
 
     if rgb_key_ids := get_rpc_key_ids(coordinator.device.status, "rgb"):
-        # RGB mode remove light & RGBW entities, add RGB entity
-        async_remove_shelly_rpc_entities(
-            hass, LIGHT_DOMAIN, coordinator.mac, [*light_keys, "rgbw:0"]
-        )
         async_add_entities(RpcShellyRgbLight(coordinator, id_) for id_ in rgb_key_ids)
-        return
 
     if rgbw_key_ids := get_rpc_key_ids(coordinator.device.status, "rgbw"):
-        # RGBW mode remove light & RGB entities, add RGBW entity
-        async_remove_shelly_rpc_entities(
-            hass, LIGHT_DOMAIN, coordinator.mac, [*light_keys, "rgb:0"]
-        )
         async_add_entities(RpcShellyRgbwLight(coordinator, id_) for id_ in rgbw_key_ids)
```

Comment:
If its possible to have multiple entities added here we should call `async_add_entities` only once since it creates a task under the hood and we only want one task.

Alternatively if its not possible we could change this to an `if/elif/elif/elif` block to show its only one